### ISSUE TITLE
Unable to parse incoming Flexible content, data should be an array

### ISSUE THREAD

**Issue by ZsarkZ** [whitecube/nova-flexible-content#360]
I have two models for example a model and a description model with a one to one relationship
when i try to create model and description model at the same time i get this error, here is my code
`//main model
public function fields(NovaRequest $request)
    {
        return [
            ID::make()->sortable(),
            Text::make('Name')->rules('required', 'max:255'),
            Slug::make('Slug')
                ->creationRules('unique:countries,slug')
                ->updateRules('unique:countries,slug,{{resourceId}}')
                ->onlyOnForms()
                ->from('Name'),
            Text::make('Local Name')->rules('required', 'max:255'),
            Text::make('Code')->rules('required', 'max:10'),
            Boolean::make('Visibility'),
            HasOne::make('Country Decription De', 'decriptionDe', \App\Nova\CountryDescriptionDe::class),
        ];
    }
// description model
public function fields(NovaRequest $request)
    {
        return [
            ID::make()->sortable(),
            BelongsTo::make('Country'),
            Flexible::make('Content')
                ->addLayout('Content section', 'content', [
                    Text::make('Title'),
                    Markdown::make('Content')
                ]),
        ];
    }
`
file:///home/sark/Ð—Ð°Ð³Ñ€ÑƒÐ·ÐºÐ¸/1656999450716.png
but if you create a model through the description model, then  work well

![](file:///home/sark/%D0%97%D0%B0%D0%B3%D1%80%D1%83%D0%B7%D0%BA%D0%B8/1657000803307.png)

**Comment by toonvandenbos** [comment]
Are you sure this problem is related to this package ? Could you provide the error you're encountering ? Thanks.

**Comment by ZsarkZ** [comment]
> Are you sure this problem is related to this package ? Could you provide the error you're encountering ? Thanks.

exception: "Exception"
file: "/home/sark/www/test/vendor/whitecube/nova-flexible-content/src/Flexible.php"
line: 394
message: "Unable to parse incoming Flexible content, data should be an array."

I think the problem with HasOne nova field. if i used HasOne nova show two form for one time main form and hasone model form and i think that Flexible film don't understand what form it need

**Comment by toonvandenbos** [comment]
Indeed, it seems to me the HasOne field does not provide the correct data structure to the flexible field. Can you debug and dump what the field is receiving when used inside a HasOne field ?

**Comment by toonvandenbos** [comment]
Alright, can you open your vendor files and add a dump/log [after this line](https://github.com/whitecube/nova-flexible-content/blob/70e1b77fb84fb0cc9aaf9362b28a597c0d1e7788/src/Flexible.php#L392) in order to get a glimpse of what the incoming request data looks like (the contents of `$value`) ?

**Comment by ZsarkZ** [comment]
> 



> Alright, can you open your vendor files and add a dump/log [after this line](https://github.com/whitecube/nova-flexible-content/blob/70e1b77fb84fb0cc9aaf9362b28a597c0d1e7788/src/Flexible.php#L392) in order to get a glimpse of what the incoming request data looks like (the contents of `$value`) ?

[2022-07-18 12:55:37] laravel.WARNING: Return type of Whitecube\NovaFlexibleContent\Layouts\Layout::offsetUnset($offset) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /home/sark/www/test/vendor/whitecube/nova-flexible-content/src/Layouts/Layout.php on line 504  
[2022-07-18 12:55:37] local.INFO: null  
[2022-07-18 12:55:37] local.INFO: "[{\"layout\":\"flex_content\",\"key\":\"M0t8gmxEkF1YwKbE\",\"attributes\":{\"M0t8gmxEkF1YwKbE__top_description\":\"<p>test<\/p>\",\"M0t8gmxEkF1YwKbE__bottom_description\":\"<p>test<\/p>\"}}]"  
[2022-07-18 12:55:37] local.ERROR: Unable to parse incoming Flexible content, data should be an array. {"userId":1,"exception":"[object] (Exception(code: 0): Unable to parse incoming Flexible content, data should be an array. at /home/sark/www/test/vendor/whitecube/nova-flexible-content/src/Flexible.php:395)

**Comment by toonvandenbos** [comment]
Hmmm, besides the weird warning (which version of PHP are you using ?), it seems the field is receiving an unparsed JSON string. Meaning the HasOne field does not forward the parsed value. There is not much we can do about it right now, I'm afraid. Did you open an issue on Nova's repository ?

**Comment by ZsarkZ** [comment]
i use php 8.0 and i didn't open an issue on Nova's repository

**Comment by ZsarkZ** [comment]
> Hmmm, besides the weird warning (which version of PHP are you using ?), it seems the field is receiving an unparsed JSON string. Meaning the HasOne field does not forward the parsed value. There is not much we can do about it right now, I'm afraid. Did you open an issue on Nova's repository ?

this is what they told me on nova repository

HasOne field has the correct structure and it is working as expected. It could just be the field isn't compatible with nova-flexible-content package.

**Comment by toonvandenbos** [comment]
In that case, maybe you should parse the field's value yourself using one of the available callbacks.

**Comment by Kreshnik** [comment]
I am encountering the same issue, has there by chance been any progress/update?
